sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -6,6 +6,19 @@
 using Azure::Core::Context;
 using namespace Azure::Core::Http;
 
+std::shared_ptr<HttpTransport> Azure::Core::Http::Details::GetTransportAdapter()
+{
+  // The order of these checks is important so that WinHttp is picked over Curl on Windows, when
+  // both are defined.
+#if defined(BUILD_TRANSPORT_CUSTOM_ADAPTER)
+  return ::AzureSdkGetCustomHttpTransport();
```

Comment:
Maybe one of these (don't really like having `Sdk` in the type name), but I don't feel too strongly about it:
- `AzureGetCustomHttpTransport`
- `AZ_GetCustomHttpTransport`
- `AZGetCustomHttpTransport`
```suggestion
  return ::`AzureGetCustomHttpTransport`?();
```